projects/evm-compatible-bridge/design.md

```diff
@@ -0,0 +1,303 @@
+
+Also, the operation when sending the tokens to the user will not be to mint tokens to the user account but to call the token contract to send the tokens to the user account.
+
+## Federation service
+
+This service will be the main actor in the bridge, it will listen for events on the EVM bridge and on Hathor to create the transactions and calls necessary to cross tokens from one side of the bridge to another.
+
+For increased security we will have a certain number of copies of the service running (i.e. M) each with their own private key,
+And any operation on both sides of the bridge will require a minimum number of participants to agree to the transaction (i.e. N).
+
+On the EVM side this can be achieved with a federation style voting system, under which some operations can be proposed by any participant of the federation but will require a minimum number of voters on this proposal so it can actually be executed.
+
+On Hathor this is more strict, since the best way to ensure the minimum number of participants agree is to use a MultiSig wallet, which requires N out of M participants to sign a transaction before it is even recorded on the ledger.
+The admins of this wallet cannot simply remove a participant from the MultiSig since it would change the actual wallet so any changes on the participants or the number of minimum signatures would require a redeploy of all instances of the service.
+
+The EVM provides events (or logs) which can be used to communicate a new proposal and request for signatures, but for the MultiSig wallet on Hathor the communication needs to be handled in a different way.
+To achieve this we introduce a coordinator service, which will listen for transactions in the MultiSig wallet and coordinate which utxos will be spent for each transaction.
+This coordinator service will also provide some APIs so admins can issue commands to the federation and so admins can check the state of the bridge.
+
+## Hathor MultiSig
+
+Hathor MultiSig standard relies on a pay-to-script-hash (P2SH) model, this means that for any data to be written on-chain (e.g. transactions, token create/mint/melt operations) the signatures for this "write" must already be collected.
+
+To handle operations on the MultiSig wallet each participant will run a copy of a headless wallet initialized with its seed and MultiSig configuration.
+
+For the coordinator service to be notified that a transaction was sent to the "federation" MultiSig wallet we will use the websocket extension of the headless which enables real time notification of a received transaction.
+The coordinator is not a participant of the federation so his headless wallet will be initialized with the MultiSig configuration but without a seed, this does not mean it is a readonly wallet since we will use it to send transactions with the collected signatures.
```

Comment:
Yes, the multisig configuration can allow for a wallet to generate the addresses an listen for transactions.
This way we can have an instance of the headless that does not have the seed, only works to coordinate the other instances that have a seed.

It will be able to generate the addresses and listen for transactions, but it will not be able to sign or modify the transactions being sent.
Since we use a `SIGHASH_ALL` strategy to signing transactions the coordinator will not be able to change the details of the transaction and since it does not have the private key to any of the participants it won't be able to sign any transaction.

Reply:
Ok. I believe the text should be updated for clarity. As I said, I was confused when I first read it and it took me a while to understand.